### PR TITLE
Documentation changes and addition for NCTools v 2022.02

### PR DESCRIPTION
**Pull request by ngs333**
This PR creates a docs folder, initially containing a documentation files on extreme fregrid, and on 
combining restart and history files ( a paragraph initially in the README.md).

The README.md file was also changed, most notable modifying a section on the documentation and adding a section 
concerning user contributions, but there were other changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,6 +69,22 @@ autotools configure command may need to be specified. This includes the
 and the ```--enable-venv``` option to build a Python venv containing the
 Ocean Model Grid Generator script and all python dependencies.
 
+## Contributing to NCTools
+
+The CNTools project encourages contributions from external parties.
+This includes from bug fixes and  enhancements to existing code, to adding
+entire external projects as submodules for distribution and installation
+long with NCTools. Contribution requirements include :
```
Comment (bensonr) [inline README.md:77]: typo "long" -> "along"

File: README.md [hunk 2]
```diff
@@ -65,6 +69,22 @@ autotools configure command may need to be specified. This includes the
 and the ```--enable-venv``` option to build a Python venv containing the
 Ocean Model Grid Generator script and all python dependencies.
 
+## Contributing to NCTools
+
+The CNTools project encourages contributions from external parties.
+This includes from bug fixes and  enhancements to existing code, to adding
```
Comment (bensonr) [inline README.md:75]: "This includes from" wording is a little awkward.

File: README.md [hunk 3]
```diff
@@ -176,26 +196,23 @@ and upgrading to a sufficiently recent compiler version is recommended in this c
...
-the candidate file first.
-
-If it returns 0, it's land compressed file that needs:
-1. combine-ncc inputfile1 inputfile2 outputfile
-2. decompress-ncc inputfile outputfile
-
-If is-compressed returns non-zero, it's a regular distributed file that gets:
-1. mppnccombine outputfile inputfile1 inputfile2 ...
-
+Documentation for on using individual applications (tools) may be obtained by running
```
Comment (bensonr) [inline README.md:199]: "Documentation for on" wording is awkward.

**Review by bensonr — COMMENTED** [review]
Also, is there a note in the documenation about using -64 with mppnccombine, or have we finally changed the default behavior to always use that option?

File: README.md (hunk 1 above)
Comment (ngs333) [reply README.md:77]: @bensonr , @ceblanton , @underwoo 
Fixed typos mentioned above,  made some text simplifications, and updated the flow of sections a little. Let me know if more changes are needed.
Thanks

File: README.md [hunk 4]
```diff
@@ -65,6 +69,22 @@ autotools configure command may need to be specified. This includes the
 and the ```--enable-venv``` option to build a Python venv containing the
 Ocean Model Grid Generator script and all python dependencies.
 
+## Contributing to NCTools
+
+The CNTools project encourages contributions from external parties.
```
[the remark this answers is not in the document]
Comment (ngs333) [reply README.md:74]: fixed.

File: README.md (hunk 1 above)
Comment (ngs333) [reply README.md:77]: fixed

File: README.md (hunk 2 above)
Comment (ngs333) [reply README.md:75]: fixed

File: README.md (hunk 3 above)
Comment (ngs333) [reply README.md:199]: fixed

**Comment by ngs333** [comment]
> Also, is there a note in the documenation about using -64 with mppnccombine, or have we finally changed the default behavior to always use that option?

@bensonr , @ceblanton , @underwoo 
The new README.md at the end of the "User Documentation" section near the top suggest to run the individual tool without arguments or with -h or --help in order to get user documentation. For mppnccombine, the first two options will output the documentation , and part of what it  displays is:
```
  -64   Create netCDF output files with the 64-bit offset format.
  -n4   Create netCDF output files in NETCDF4_CLASSIC mode (no v4 enhanced features).
```

**Pull request merged**